src/common/xo-week-charts/index.js

```diff
@@ -0,0 +1,414 @@
+      .x(elem => this.x(elem.date))
+      .y(elem => this.y(elem.value))
+
+    // Tooltip ---------------------------------------------
+
+    svg
+      .append('rect')
+        .attr('class', 'mouse-area')
+        .style(MOUSE_AREA_STYLE)
+        .on('mousemove', this._handleMouseMove)
+
+    const hover = svg
+      .append('g')
+        .attr('class', 'hover-container')
+        .style('pointer-events', 'none')
+
+    hover
+      .append('line')
+        .attr('class', 'hover-line')
+        .attr('y1', 0)
+        .style(HOVER_LINE_STYLE)
+
+    hover
+      .append('text')
+        .attr('class', 'hover-text')
+        .style(HOVER_TEXT_STYLE)
+
+    this._draw()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { props } = this
+
+    if (!propsEqual(
+      props,
+      nextProps,
+      [ 'chartHeight', 'chartWidth', 'data', 'maxValue' ]
+    )) {
+      this._draw(nextProps)
+    }
+
+    if (props.tooltipX !== nextProps.tooltipX) {
+      this._updateTooltip(nextProps.tooltipX)
+    }
+  }
+
+  render () {
+    return <div ref='chart' />
+  }
+}
+
+// ===================================================================
+
+@propTypes({
+  chartHeight: propTypes.number,
+  series: propTypes.arrayOf(
+    propTypes.shape({
+      data: propTypes.arrayOf(
+        propTypes.shape({
+          date: propTypes.number.isRequired,
+          value: propTypes.number.isRequired
+        })
+      ).isRequired,
+      objectId: propTypes.string.isRequired
+    })
+  ).isRequired,
+  valueRenderer: propTypes.func
+})
+export default class XoWeekCharts extends Component {
+  _handleResize = () => {
+    const { container } = this.refs
+    this.setState({
+      chartsWidth: container && container.offsetWidth || undefined
+    })
+  }
+
+  _handleTooltipChange = x => {
+    this.setState({ tooltipX: x })
+  }
+
+  _updateScale = (useScale, series = this.props.series) => {
+    let max
+
+    if (useScale) {
+      max = 0
+      forEach(series, serie => {
+        forEach(serie.data, elem => {
+          max = Math.max(elem.value, max)
+        })
+      })
+    }
+
+    this.setState({
+      maxValue: max
+    })
+  }
+
+  componentDidMount () {
+    window.addEventListener('resize', this._handleResize)
+    this._handleResize()
+  }
+
+  componentWillMount () {
+    this.setState({
+      tooltipX: 0
+    })
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('resize', this._handleResize)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { series } = nextProps
+
+    if (this.props.series !== series) {
+      this.setState({ tooltipX: 0 })
+      this._updateScale(this.state.maxValue !== undefined, series)
+    }
+  }
+
+  render () {
+    const {
+      props,
+      state: {
+        chartsWidth,
+        maxValue,
+        tooltipX
+      }
+    } = this
+
+    return (
+      <div>
+        <div>
+          <p className='m-t-1'>
+            {_('weeklyChartsScaleInfo')}
+            {' '}
```

Comment:
Useless because it will be removed anyway.
Use `&nbsp;` instead if you want a white space.